testcases/PowerNVDump.py

```diff
@@ -1324,6 +1330,262 @@ def runTest(self):
+        '''
+        self.get_watchdog_timeout_value()
+        self.get_watchdog_action_value()
+
+        if (self.system_timeout_value == self.script_timeout) and (self.watchdog_mode == self.script_action):
+            return True
+        else:
+            self.fail("Timeout and action values of watchdog module"
+                      "are incorrect,Please check logs")
+
+    def get_watchdog_timeout_value(self, i_module):
+        '''
+        Funtion to get timeout value of a watchdog_module
+        that set by script.
+
+        :params i_module: watchdog module i.e "pseries_wdt"
+        :rtype int
+        '''
+        cmd = f"cat /sys/module/{i_module}/parameters/timeout"
+        output = self.cv_HOST.host_run_command(cmd)
+        self.system_timeout_value = int(output[0])
+        return self.system_timeout_value
+
+    def get_watchdog_action_value(self, i_module):
+        '''
+        Funtion to get action value of a watchdog_module
+        that set by script.
+
+        :params i_module: watchdog module i.e "pseries_wdt"
+        :rtype int
+        '''
+        cmd = f"cat /sys/module/{i_module}/parameters/action"
+        output = self.cv_HOST.host_run_command(cmd)
+        self.watchdog_mode = int(output[0])
+        return self.watchdog_mode
+
+    def module_load(self):
+        '''
+        Funtion to load watchdog module.
+        '''
+        self.cv_SYSTEM.cv_HOST.host_load_module("pseries_wdt")
+
+    def module_unload(self, i_module):
+        '''
+        This function will unload the module using modprobe
+        and validates module unload.
+        '''
+        try:
+            self.cv_HOST.host_run_command("modprobe -r %s" % i_module)
+        except CommandFailed as c:
+            l_msg = "Error in unloading the module %s, modprobe -r failed: %s" % (
+                i_module, str(c))
+            raise OpTestError(l_msg)
+        if self.cv_HOST.host_check_module_presence(i_module):
+            raise OpTestError(f"{i_module} module still present even after unload,Please check logs")
+
+    def check_module_support(self):
+        '''
+        Function to check the watchdog module is supported with the current kernel
+        if not supported, none of the test cases executed.
+
+        :return : True if supported else False
+        :rtype : boolean
+        '''
+        cmd = "find /lib/modules/$(uname -r) -type f -name '*.ko*' | grep pseries-wdt"
+        try:
+            self.cv_HOST.host_run_command(cmd)
+            return True
+
+        except CommandFailed as c:
+            msg = " Watchdog module is not supported in this kernel, Please check."
+            raise OpTestError(msg)
+
+    def watchdog_run_command(self):
+        '''
+        This funtion runs the watchdog event trigger command.
+        '''
+        if self.validate_timeout_and_action:
+            cmd = "./watchdog-countdown"
+            self.c.pty.sendline(cmd)
```

Comment:
when a command is send over raw_pty than, you should expect some value .. for it to  continue or print the previous logs

Reply:
I think watchdog_run_command() is not used anywhere ...should you remove this method